scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+            if (sftpMode.equals("PUT")) {
+                throw new IllegalArgumentException("Please specify a local relative path. Empty value is not allowed.");
+            }
+            //default value is getlocalspace() because it will always be writable and moreover can be used to transfer files to another data space(global, user)
+            sftpLocalRelativePath = getLocalSpace();
+        }
+        if (args.containsKey("sftpRemoteRelativePath")) {
+            sftpRemoteRelativePath = args.get("sftpRemoteRelativePath").toString();
+        }
+
+        if (args.containsKey("sftpUsername")) {
+            sftpUsername = args.get("sftpUsername").toString();
+        }
+
+        if (sftpUsername == null) {
+            throw new IllegalArgumentException("Username is required to access an SFTP server");
+        }
+
+        // This key is used for logs and for getting the password from 3rd party credentials.
+        sftpUrlKey = "sftp://" + sftpUsername + "@" + sftpHostname + ":" + sftpPort;
+
+        sftpPassword = getThirdPartyCredential(sftpUrlKey);
+
+        if (sftpPassword == null) {
+            throw new IllegalArgumentException("Please add your sftp password to 3rd-party credentials under the key :\"" +
+                                               sftpUrlKey + "\"");
+        }
+    }
+
+    @Override
+    public Serializable execute(TaskResult... results) throws IOException, JSchException, SftpException {
+
+        List<String> filesRelativePathName = null;
+
+        ChannelSftp channelsftp = null;
+        Session session = null;
+        Channel channel = null;
+
+        try {
+            JSch jsch = new JSch();
+            session = jsch.getSession(sftpUsername, sftpHostname, sftpPort);
+            session.setPassword(sftpPassword);
+            Properties config = new Properties();
+            config.put("StrictHostKeyChecking", "no");
+            session.setConfig(config);
+            // create sftp Session
+            getOut().println("Connecting to " + sftpUrlKey);
+            session.connect();
+            // open sftp Channel
+            channel = session.openChannel("sftp");
+            channel.connect();
+            channelsftp = (ChannelSftp) channel;
+
+            switch (sftpMode) {
+                //ftp mode is get
+                case "GET":
+                    getOut().println("BEGIN DOWNLOAD FROM SFTP SERVER: " + sftpUrlKey);
+                    // recursive folder content download from sftp server
+                    filesRelativePathName = recursiveFolderDownload(channelsftp,
+                                                                    sftpRemoteRelativePath,
+                                                                    sftpLocalRelativePath);
+                    getOut().println("END DOWNLOAD FROM SFTP");
+                    break;
+
+                //ftp mode is put
+                case "PUT":
```

Comment:
We can use the constant here instead of "PUT".

Reply:
Agree